experimental/LieAlgebras/src/RootSystem.jl

```diff
@@ -21,7 +23,13 @@
 
     R = new(mat)
     R.positive_roots = map(r -> RootSpaceElem(R, r), pos_roots)
+    R.positive_roots_map = Dict(
+      (coefficients(root), ind) for (ind, root) in enumerate(R.positive_roots::Vector{RootSpaceElem})
+    )
```

Comment:
pre-computing this and using this in `is_root` instead of a linear search is the main speed improvement of this PR